### PR TITLE
Console login default namespace

### PR DESCRIPTION
**Pull request by Jondeen**
I've added a few functionalities which may be of interest. 

1) Enabling a console version of the token handling. I mostly work in remote consoles, and then it becomes difficult to open a functional js-enabled browser from the shell. -manual-input enables copy-pasting the token request url, and pasting back the authetication token upon redirect. I'm not sure if it is possible to make this more elegant from the server side - if so I'd love to know how to adapt to it.

2) Enabled adding a default namespace to the kube config. Super handy if your work is mostly related to one specific namespace.

Both functionalities store their parameters in kubedconf so it's compatible with a -renew command.

### CODE REVIEW

**Comment by Jondeen** [comment]
I rewrote some hard-coded path changes to make Travis happy (build has been failing for over a year), but double check that these changes are compatible with any downstream systems. There seem to be some inconsistencies in naming convention (`UNINETT` vs `uninett` vs `Uninett`) and I'm not sure which one is correct.

File: main.go [hunk 1]
```diff
@@ -89,16 +94,27 @@ func main() {
 	}
 
 	log.Info("Requesting Access Token from Dataporten")
+	err = nil
+	token := ""
+
+	// Manually fetch token if browser is unavailable from console:
+	if (cluster.ManualInput) {
+		fmt.Println("Open a browser and navigate to " + authURL + "?response_type=token&client_id=" + cluster.ClientID + ", then report the return token below.")
```
Comment (gurvindersingh) [inline main.go]: Would be good to inform user that here they need to copy token from the url itself. The value is in the request param `access_token`.

File: token.go [hunk 2]
```diff
@@ -71,9 +71,5 @@ func getToken(port int) (string, error) {
 
 	token := <-done
 
-	err := srv.Close()
-	if err != nil {
-		return token, errors.Wrap(err, "Error shutting down server")
-	}
```
Comment (gurvindersingh) [inline token.go]: Logic for deletion is that, anyway program will exit and server will be closed or I miss something here.

**Comment by gurvindersingh** [comment]
Thanks @Jondeen for the PR. Added some small comments. Overall it looks good. Regarding Dataporten response, there is no good way to get the token in user friendly way currently which I am aware of.

Between which institution you work for, as this tool is bit targeted mainly for UH-sector in Norway :)

**Comment by Jondeen** [comment]
So I fixed the erroneously deleted closure of the http-server instance, and also made a bit more user friendly way of extracting the access_token from the redirect-url.

Travis now fails somehow, though after the push a lot of glitches spun up on the Github/Travis web ui, so I'm wondering if this is an issue on the Github/Travis-part and not in the code itself. A manually triggered Travis rebuild would be advised.

EDIT: Closing/Reopening the PR triggered a rebuild from Travis, and as expected, it worked fine this time around.

File: token.go (hunk 2 above)
Comment (Jondeen) [reply token.go]: Seems like Github dropped my previous comment here - but this was wrongfully dropped in my first commit (I think) due to me working on a Go version from before Close() was implemented (1.7). I fixed the drop in the following commit.

File: main.go [hunk 3]
```diff
@@ -99,10 +99,23 @@ func main() {
 
 	// Manually fetch token if browser is unavailable from console:
 	if (cluster.ManualInput) {
-		fmt.Println("Open a browser and navigate to " + authURL + "?response_type=token&client_id=" + cluster.ClientID + ", then report the return token below.")
-		fmt.Print("access_token from redirect url: ")
-		token, err = bufio.NewReader(os.Stdin).ReadString('\n')
-		token = token[:len(token)-1]
+		fmt.Println("Open a browser and navigate to " + authURL + "?response_type=token&client_id=" + cluster.ClientID)
+        fmt.Println("After authentication, you are redirected to an invalid URL. Copy/paste this url below:")
+		fmt.Print("Redirect-URL: ")
```
Comment (gurvindersingh) [inline main.go]: `Redirected-URL:`

File: main.go [hunk 4]
```diff
@@ -99,10 +99,23 @@ func main() {
...
+        token_url_string := ""
+		token_url_string, err = bufio.NewReader(os.Stdin).ReadString('\n')
+        if err != nil {
+                log.Fatal("Something disastrous happened while getting input from console ", err)
```
Comment (gurvindersingh) [inline main.go]: `input from console, please run kubed again"`

**Review by gurvindersingh — CHANGES_REQUESTED** [review]
Couple of small nits and I am ready to merge.

**Review by gurvindersingh — APPROVED** [review]

**Pull request merged**